File: codes/cpp/inheritance/multiple/multiple1.cpp
#include <iostream>
#include <string>

class Person {
public:
    Person(){
        std::cout << "I am a person" << std::endl;
    }
};

class Machine {
public:
    Machine(){
        std::cout << "I am a machine" << std::endl;
    }
};

class Android: public Machine, public Person {
public:
    Android(){
        std::cout << "So, I am an android" << std::endl;
    }
};

class Cyborg: public Person, public Machine {
public:
    Cyborg(): Machine(), Person(){
        std::cout << "So, I am a cyborg" << std::endl;
    }
};


int main()
{
    Person p;
    std::cout << "=====================" << std::endl;
    Machine m;
    std::cout << "=====================" << std::endl;
    Android a;
    std::cout << "=====================" << std::endl;
    Cyborg c;
    std::cout << "=====================" << std::endl;
}

File: codes/php/inheritance/multiple/multiple1.php
<?php

interface iPerson { public function info(); }

interface iMachine { public function info(); }

trait tPerson {
    public function info() { echo "I am a person\n"; }
}

trait tMachine {
    public function info() { echo "I am a machine\n";}
}

trait tConstruct {
    public function __construct() {
        if (get_parent_class()) { parent::info(); }
        $this->info();
        $this->moreInfo();
    }
    public function moreInfo(){}
}

class Person implements iPerson { use tPerson, tConstruct;}

class Machine implements iMachine { use tMachine, tConstruct; }

class Android extends Machine implements iPerson {
    use tPerson, tConstruct;
    public function moreInfo(){ echo "So, I am an android\n";}
}

class Cyborg extends Person implements iMachine {
    use tMachine, tConstruct;
    public function moreInfo(){ echo "So, I am a cyborg\n";}
}


$p = new Person();
echo "=====================\n";
$m = new Machine();
echo "=====================\n";
$a = new Android();
echo "=====================\n";
$c = new Cyborg();
echo "=====================\n";



?>
